### DISCUSSION TITLE
Use DEFAULT keyword in an UPDATE statement

### DISCUSSION THREAD

**Discussion by rharish101** [SeaQL/sea-orm#1220] [Q&A] [answered]
I would like to set a column value to its default value. However, I can't seem to find it in the docs.

For example, consider a PostgreSQL table created with the following schema:
```sql
CREATE TABLE IF NOT EXISTS table_name (
	index_name INT NOT NULL,
	column_name TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
	PRIMARY KEY (index_name)
);
```

Now, I'd like to run the following operation:
```sql
UPDATE table_name SET column_name = DEFAULT WHERE index_name = 1;
```

If I do something like this:
```rust
table_name::Entity::update_many()
    .col_expr(table_name::Column::IndexName, Expr::value(1))
    .col_expr(table_name::Column::ColumnName, Expr::value("DEFAULT"))
    .exec(db)
    .await?;
```
it doesn't work, since the `"DEFAULT"` being used here is interpreted as a string value (while the column `column_name` is for timestamps), not as the command `DEFAULT`. I realize that I could just use `Expr::current_timestamp()` here, since that's my default value, but isn't there support for the `DEFAULT` keyword?

Note that here, I'm using `sea-orm-codegen` to generate the entity `src/entities/table_name`:
```rust
//! SeaORM Entity. Generated by sea-orm-codegen 0.10.2

use sea_orm::entity::prelude::*;

#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq)]
#[sea_orm(table_name = "table_name")]
pub struct Model {
    #[sea_orm(primary_key, auto_increment = false)]
    pub index_name: i32,
    pub column_name: DateTime,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}

impl ActiveModelBehavior for ActiveModel {}
```

**Comment by rharish101** [comment, marked as answer]
I got it to work by using `Expr::cust("DEFAULT")`. However, shouldn't something like `Expr::default_value()` already exist?

**Reply by tyt2y3** [reply]
Please avoid rhetorical question. No, it does not exist.
We have "insert or default" in SeaQuery, but not "update default". I think we can add "Default" to [Keyword](https://docs.rs/sea-query/latest/sea_query/types/enum.Keyword.html)
